src/hotspot/share/opto/macro.cpp

```diff
@@ -1672,7 +1672,7 @@ PhaseMacroExpand::initialize_object(AllocateNode* alloc,
     transform_later(mark_node);
   }
   rawmem = make_store(control, rawmem, object, oopDesc::mark_offset_in_bytes(), mark_node, TypeX_X->basic_type());
-
+  rawmem = make_store(control, rawmem, object, oopDesc::nklass_offset_in_bytes(), klass_node, T_METADATA);
   rawmem = make_store(control, rawmem, object, oopDesc::klass_offset_in_bytes(), klass_node, T_METADATA);
```

Comment:
This looks patently weird. What do we store to, `nklass` or `klass`? Why both here?

Reply:
The Klass* will be eliminated in the subsequent PR. Until then, we have to keep both.

Reply:
I believe I can safely remove the store to the nklass_offset. The narrow Klass* in the header is already initialized by storing the initial mark word. I only put the store to nklass_offset there to set-up a memory edge to subsequent load-klass. However, the initializer should always emit a membar, which provides the required memory ordering between the initializing store of the narrowKlass* and subsequent loads.